### PR TITLE
Backport #11080 and #11081 (3.3)

### PR DESCRIPTION
**Pull request by bernd**
Please let me merge the PR after approval. Thanks!

### CODE REVIEW

File: graylog2-server/src/main/java/org/graylog2/security/UserSessionTerminationService.java [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+import com.google.auto.value.AutoValue;
+import com.google.common.util.concurrent.AbstractIdleService;
+import org.apache.shiro.mgt.DefaultSecurityManager;
+import org.apache.shiro.session.ExpiredSessionException;
+import org.apache.shiro.session.Session;
+import org.apache.shiro.session.mgt.DefaultSessionKey;
+import org.apache.shiro.session.mgt.SessionManager;
+import org.graylog2.plugin.ServerStatus;
+import org.graylog2.plugin.cluster.ClusterConfigService;
+import org.graylog2.users.events.UserChangedEvent;
```
Comment (thll) [inline graylog2-server/src/main/java/org/graylog2/security/UserSessionTerminationService.java:31]:
```suggestion
```

File: graylog2-server/src/main/java/org/graylog2/security/UserSessionTerminationService.java [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.io.Serializable;
+import java.util.Optional;
+
+/**
+ * This service checks on startup if all user sessions should be terminated. This can be requested by bumping the
+ * {@code TERMINATION_REVISION} number.
+ *
+ * The service also listens for {@link UserChangedEvent} events and terminates all sessions for disabled users.
```
Comment (thll) [inline graylog2-server/src/main/java/org/graylog2/security/UserSessionTerminationService.java:44]:
```suggestion
```

**Review by thll — CHANGES_REQUESTED** [review]

File: graylog2-server/src/main/java/org/graylog2/security/UserSessionTerminationService.java (hunk 1 above)
Comment (bernd) [reply graylog2-server/src/main/java/org/graylog2/security/UserSessionTerminationService.java]: Fixed, thank you!

File: graylog2-server/src/main/java/org/graylog2/security/UserSessionTerminationService.java (hunk 2 above)
Comment (bernd) [reply graylog2-server/src/main/java/org/graylog2/security/UserSessionTerminationService.java]: Fixed, thank you!

**Review by thll — APPROVED** [review]

**Pull request merged**